### PR TITLE
[POC] Interactive cookbooks + automated testing

### PR DESCRIPTION
**Pull request by mnkiefer**
- Adds *interactive cookbook* feature to external Capire repo in order to:
  - Enable **notebook generation** of CAP Notebooks  in pipeline
  - Run **automated testing**  on them to sanitise CAP cookbook content

- We already have several notebooks enabled, which are now checked and sanitised:
  - [X] Hello, World (Node.js)
  - [X] Hello, World (Java)
  - [X] Getting started in a Nutshell (Node.js)
  - [X] Getting started in a Nutshell (Java)

### CODE REVIEW

File: .github/notebooks/html2capnb.cjs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+// RULE: Prepend `%%writefile [file]` magic command to the top of all code cells
+function addWriteFileMagic(text, language, label) {
+  const magic = language === "javascript" ? `// %%writefile "${label}"\n` : `%%writefile "${label}"\n`;
+  return magic + text;
+}
+
+// RULE: Code cells containg cds watch/ serve commands are of type 'CDS server'
+function addCDSServerLanguage(language, text) {
+  if (text.trim().includes('cds watch') ||
+      text.trim().includes('cds serve') ||
```
Comment (chgeo) [inline .github/notebooks/html2capnb.cjs:200]:
```suggestion
      text.trim().includes('cds serve') ||
      text.trim().includes('cds run') ||
```

**Review by chgeo — APPROVED** [review]
Looks good to me

**Comment by chgeo** [comment]
Can we do a test and make one of the notebooks fail, so that I can see a test error?

**Comment by mnkiefer** [comment]
> Can we do a test and make one of the notebooks fail, so that I can see a test error?

I induced an error just now (`cds init` -> `cds innit`) in the Node.js Getting started guide so you can see what it looks like.

**Comment by renejeglinsky** [comment]
> 💥 Broken cookbooks found!
> 
> Go to [test summary](https://github.tools.sap/cap/docs/actions/runs/5586151016)

The link is dead. Should be *cap-js* instead of *cap*.
I wouldn't have caught from the output what the error is. Did I just not see it (absolutely possible!)?

**Comment by mnkiefer** [comment]
> > 💥 Broken cookbooks found!
> > Go to [test summary](https://github.tools.sap/cap/docs/actions/runs/5586151016)
> 
> The link is dead. Should be _cap-js_ instead of _cap_. I wouldn't have caught from the output what the error is. Did I just not see it (absolutely possible!)?

You're right! It was still using the internal link (with GHES). I just fixed it, you should now be able to navigate to the GHA summary and see the failed Notebook result. 👍

**Comment by chgeo** [comment]
Looks good!

**Review by renejeglinsky — APPROVED** [review]

**Comment by mnkiefer** [comment]
Done testing

**Pull request closed without merging**